### PR TITLE
Add to_seconds support for tiflash

### LINKED ISSUE #4679: support `to_seconds` in TiFlash
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Support `to_seconds` in TiFlash
MySQL doc:
https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_to-seconds
TiDB implementation:
https://github.com/pingcap/tidb/blob/5889d4c4e9357a515c472fcf12bb3624fbb00f9e/expression/builtin_time.go#L7069
**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by yibin87**
Signed-off-by: yibin <huyibin@pingcap.com>

### What problem does this PR solve?

Issue Number: close #4679 

Problem Summary:

### Check List

Tests 

- [ x ] Unit test
- [ x ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Functions/FunctionsDateTime.h [hunk 1]
```diff
@@ -3277,6 +3277,42 @@ struct TiDBWeekOfYearTransformerImpl
...
+            MyTimeBase val(vec_from[i]);
+            vec_to[i] = execute(context, val, is_null);
+            vec_null_map[i] = is_null;
+            is_null = false;
+        }
+    }
+
+    static ToFieldType execute(const Context & context, const MyTimeBase & val, bool & is_null)
+    {
+        // TiDB returns normal value if one of month/day is zero for to_seconds function, while MySQL return null if either of them is zero.
```
Comment (windtalker) [inline dbms/src/Functions/FunctionsDateTime.h:3302]: Seems a TiDB bug, please fire a issue for this.

**Review by windtalker — APPROVED** [review]
LGTM

**Review by guo-shaoge — APPROVED** [review]

File: dbms/src/Functions/FunctionsDateTime.h (hunk 1 above)
Comment (yibin87) [reply dbms/src/Functions/FunctionsDateTime.h:3302]: OK，I'll fire one later.

Comment (yibin87) [reply dbms/src/Functions/FunctionsDateTime.h:3302]: https://github.com/pingcap/tidb/issues/34629

**Pull request merged**